src/eko/anomalous_dimensions/aem1.py

```diff
@@ -1,96 +1,202 @@
     """
     return as1.gamma_qg(N, nf) / constants.TR * constants.NC
 
 
 @nb.njit(cache=True)
 def gamma_phph(nf):
-    """
-    Computes the leading-order photon-photon anomalous dimension
+    r"""Compute the leading-order photon-photon anomalous dimension.
 
     Implements Eq. (2.5) of :cite:`Carrazza:2015dea`.
 
     Parameters
     ----------
-      nf : int
+    nf : int
         Number of active flavors
 
     Returns
     -------
-      gamma_phph : complex
-        Leading-order phton-photon anomalous dimension :math:`\\gamma_{\\gamma \\gamma}^{(0)}(N)`
+    gamma_phph : complex
+        Leading-order phton-photon anomalous dimension :math:`\\gamma_{\\gamma \\gamma}^{(0,1)}(N)`
     """
-
-    return 2 / 3 * constants.NC * 2 * nf
+    nu = constants.uplike_flavors(nf)
+    nd = nf - nu
+    return 4.0 / 3.0 * constants.NC * (nu * constants.eu2 + nd * constants.ed2)
 
 
 @nb.njit(cache=True)
-def gamma_ns(N, s1):
-    """
-    Computes the leading-order non-singlet QED anomalous dimension.
+def gamma_ns(N, sx):
+    r"""Compute the leading-order non-singlet QED anomalous dimension.
 
     Implements Eq. (2.5) of :cite:`Carrazza:2015dea`.
 
     Parameters
     ----------
-      N : complex
+    N : complex
         Mellin moment
-      s1 : complex
+    s1 : complex
         S1(N)
 
     Returns
     -------
-      gamma_ns : complex
-        Leading-order non-singlet QED anomalous dimension :math:`\\gamma_{ns}^{(0)}(N)`
+    gamma_ns : complex
+        Leading-order non-singlet QED anomalous dimension :math:`\\gamma_{ns}^{(0,1)}(N)`
     """
+    s1 = sx[0]
     return as1.gamma_ns(N, s1) / constants.CF
+
+
+@nb.njit(cache=True)
+def gamma_singlet(N, nf, sx):
+    r"""Compute the leading-order singlet anomalous dimension matrix.
+
+    .. math::
+        \\gamma_S^{(0)} = \\left(\begin{array}{cc}
+        0 & 0 & 0 & 0 \\
+        0 & \\gamma_{\\gamma \\gamma}^{(0,1)} & \\langle e^2 \rangle \\gamma_{\\gamma q}^{(0,1)} & \nu_u e^2_- \\gamma_{\\gamma q}^{(0,1)}\\
+        0 & \\langle e^2 \rangle\\gamma_{q \\gamma}^{(0,1)} & \\langle e^2 \rangle \\gamma_{ns}^{(0,1)} & \nu_u e^2_- \\gamma_{ns}^{(0,1)}\\
+        0 & \nu_d e^2_- \\gamma_{q \\gamma}^{(0,1)} & \nu_d e^2_- \\gamma_{ns}^{(0,1)} & e^2_\\Delta \\gamma_{ns}^{(0,1)}
+        \\end{array}\right)
+
+    Parameters
+    ----------
+    N : complex
+        Mellin moment
+    s1 : complex
+        harmonic sum :math:`S_{1}`
+    nf : int
+        Number of active flavors
+
+    Returns
+    -------
+    gamma_S_0 : numpy.ndarray
+        Leading-order singlet anomalous dimension matrix :math:`\\gamma_{S}^{(0)}(N)`
+
+    See Also
+    --------
+    gamma_ns : :math:`\\gamma_{qq}^{(0)}`
+    gamma_qg : :math:`\\gamma_{qg}^{(0)}`
+    gamma_gq : :math:`\\gamma_{gq}^{(0)}`
+    gamma_gg : :math:`\\gamma_{gg}^{(0)}`
+    """
+    e2avg = constants.e2avg(nf)
+    vue2m = constants.vue2m(nf)
+    vde2m = constants.vde2m(nf)
+    e2delta = vde2m - vue2m + e2avg
```

Comment:
since this is repeated at least twice (and potentially even more often), I wonder whether it is worth to wrap them into a single function:
`e2avg,vue2m,vde2m,e2delta = constants.collect_all_charges(nf)` (function name is subject to discussion)